userspace/falco/falco_metrics.cpp

```diff
@@ -0,0 +1,114 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "falco_metrics.h"
+
+#include "app/state.h"
+
+#include <libsinsp/sinsp.h>
+
+falco_metrics::falco_metrics(falco::app::state& state)
+{
+	falco_configuration::webserver_config webserver_config = state.config->m_webserver_config;
+	m_is_enabled = state.config->m_metrics_enabled && webserver_config.m_metrics_enabled;
+
+	if (m_is_enabled)
+	{
+		for (const auto& source_info: state.source_infos)
+		{
+			sinsp *source_inspector = source_info.inspector.get();
+			m_inspectors.push_back(source_inspector);
+			m_metrics_collectors.push_back(libs::metrics::libs_metrics_collector(source_inspector, state.config->m_metrics_flags));	
+		}
+	}
+}
+
+std::string falco_metrics::to_text() const
+{
+	if (!m_is_enabled)
+	{
+		return "";
+	}
+
+	static const char* all_driver_engines[] = {
+		BPF_ENGINE, KMOD_ENGINE, MODERN_BPF_ENGINE,
+		SOURCE_PLUGIN_ENGINE, NODRIVER_ENGINE, GVISOR_ENGINE };
+
+
+	libs::metrics::prometheus_metrics_converter converter;
+	std::string prometheus_text;
+
+	for (auto* inspector: m_inspectors)
+	{
+		for (size_t i = 0; i < sizeof(all_driver_engines) / sizeof(const char*); i++)
+		{
+			if (inspector->check_current_engine(all_driver_engines[i]))
+			{
+				prometheus_text += converter.convert_metric_to_text_prometheus("engine_name", "falcosecurity", "scap", {{"engine_name", all_driver_engines[i]}});
+				break;
+			}
+		}
+
+		const scap_agent_info* agent_info = inspector->get_agent_info();
+		const scap_machine_info* machine_info = inspector->get_machine_info();
+
+		libs::metrics::libs_metrics_collector libs_metrics_collector(inspector, 0);
+
+		prometheus_text += converter.convert_metric_to_text_prometheus("falco_version", "falcosecurity", "falco", {{"falco_version", FALCO_VERSION}});
+		prometheus_text += converter.convert_metric_to_text_prometheus("kernel_release", "falcosecurity", "falco", {{"kernel_release", agent_info->uname_r}});
+		prometheus_text += converter.convert_metric_to_text_prometheus("evt_hostname", "falcosecurity", "falco", {{"evt_hostname", machine_info->hostname}});
+
+		std::vector<metrics_v2> static_metrics;
+		static_metrics.push_back(libs_metrics_collector.new_metric("start_ts",
+											METRICS_V2_LIBBPF_STATS,
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_TIME_NS_COUNT,
+											METRIC_VALUE_METRIC_TYPE_MONOTONIC,
+											agent_info->start_ts_epoch));
+		static_metrics.push_back(libs_metrics_collector.new_metric("falco_host_boot_ts",
+											METRICS_V2_LIBBPF_STATS,
+											METRIC_VALUE_TYPE_U64,
+											METRIC_VALUE_UNIT_TIME_NS_COUNT,
+											METRIC_VALUE_METRIC_TYPE_MONOTONIC,
+											machine_info->boot_ts_epoch));
+		static_metrics.push_back(libs_metrics_collector.new_metric("falco_host_num_cpus",
```

Comment:
correction + rename to `host_num_cpus` without falco_: 
```
METRICS_V2_MISC,
METRIC_VALUE_TYPE_U32,
METRIC_VALUE_UNIT_COUNT,
METRIC_VALUE_METRIC_TYPE_NON_MONOTONIC_CURRENT,
machine_info->num_cpus));
```

Reply:
Done